kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ContainerConfiguration.java

```diff
@@ -325,12 +346,22 @@ public ContainerConfiguration build() {
             result.isFrameworkManaged = this.isFrameworkManaged;
             result.containerLoggerParameters = this.containerLoggerParameters;
             result.containerLoggingType = this.containerLoggingType;
-            result.registryCredentials = requireNonNull(this.registryCredentials,
-                    "Request Registry Credentials object cannot be null");
-            result.imageDownloadTimeoutSeconds = this.imageDownloadTimeoutSeconds;
+
```

Comment:
Is it possible/difficult to remove the additional private fields and change the old methods so that they modify the imageConfig field instead? Maybe changing the type of the imageConfig to ImageConfigurationBuilder can help.. in this way the old methods could just call the corresponding ones of the builder.. To make things easier, setImageConfiguration() could also be changed to received a ImageConfigurationBuilder instead of an ImageConfig.. then the build() method could invoke ImageConfigurationBuilder.build() to get the actual ImageConfiguration.. what do you think?

Reply:
I like that a lot. Reduces complexity by getting rid of the IF and allows the end-user to "edit" an imported image config. Making these changes now.